### PR TITLE
[TT-7151] Fix the panic while loading Go-plugin

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR recovers the possible panics happening during load of a Go-plugin

### CODE REVIEW

File: gateway/mw_go_plugin.go [hunk 1]
```diff
@@ -139,6 +139,13 @@ func (m *GoPluginMiddleware) loadPlugin() bool {
 		m.Path = m.goPluginFromTykVersion(VERSION)
 	}
 
+	defer func() {
+		if e := recover(); e != nil {
+			err = fmt.Errorf("%v", e)
+			log.WithError(err).Error("Recovered from panic while loading Go-plugin")
```
Comment (buger) [inline gateway/mw_go_plugin.go:145]: Should it have explicit "return false" for readability?

Comment (furkansenharputlu) [reply gateway/mw_go_plugin.go:145]: I think it doesn't make sense because we call an inner function.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4416/merge
Commit: 9f38f1d8fc8ffc8470781b0e336067addcdb9e3c 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3523763955)

File: gateway/mw_go_plugin.go (hunk 1 above)
Comment (titpetric) [reply gateway/mw_go_plugin.go]: I would address this as a separate PR, function could return error, and even the current return is unread in parent:

```
// global go plugins
if m.Path != "" && m.SymbolName != "" {
        m.loadPlugin() // no check, no log
        return true
}
```

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4416/merge
Commit: 7454b79c97de2ecd022746be50557b921bdfac6d 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3531761635)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4416/merge
Commit: 7454b79c97de2ecd022746be50557b921bdfac6d 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3531761635)

File: gateway/mw_go_plugin.go [hunk 2]
```diff
@@ -139,6 +139,13 @@ func (m *GoPluginMiddleware) loadPlugin() bool {
 		m.Path = m.goPluginFromTykVersion(VERSION)
 	}
 
+	defer func() {
+		if e := recover(); e != nil {
+			err = fmt.Errorf("%v", e)
```
Comment (tbuchaillot) [inline gateway/mw_go_plugin.go:144]: just a little bit of comment here - Do we want to print the whole stack trace? Wouldn't be better to just print the plugin name/path that's failing?

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4416/merge
Commit: 9b884c2eafe13b6277a7566972fc18cf900e7d9e 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3549602738)

**Review by tbuchaillot — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4416/merge
Commit: f74ada96b91c9810cf3c8ad87799788870d19d74 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3572596048)

**Review by titpetric — APPROVED** [review]

**Pull request merged**